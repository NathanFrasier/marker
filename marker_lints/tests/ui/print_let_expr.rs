fn main() {
    let _str = r#"Hello"#;
    let _byte_str = b"Hello\n";
    let _int = 17_i32;
    let _float = 0.000015;
    let _char = 'h';
    let _hex_int = 0xcafe;
    let _byte = b'D';
    let _block_int = { -3 };
}
