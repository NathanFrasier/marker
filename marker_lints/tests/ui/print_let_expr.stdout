StrLit(
    StrLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        is_raw: true,
        str_data: Sym(
            SymbolId(..),
        ),
    },
)

StrLit(
    StrLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        is_raw: false,
        str_data: Bytes(
            [
                72,
                101,
                108,
                108,
                111,
                10,
            ],
        ),
    },
)

IntLit(
    IntLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        value: 17,
        suffix: Some(
            I32,
        ),
    },
)

FloatLit(
    FloatLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        value: 1.5e-5,
        suffix: None,
    },
)

CharLit(
    CharLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        value: 'h',
    },
)

IntLit(
    IntLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        value: 51966,
        suffix: None,
    },
)

IntLit(
    IntLitExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        value: 68,
        suffix: None,
    },
)

Block(
    BlockExpr {
        data: CommonExprData {
            _lifetime: PhantomData<&()>,
            id: ExprId(..),
            span: SpanId(..),
        },
        stmts: [],
        expr: None,
    },
)

